modal-go/sandbox.go

```diff
@@ -168,21 +170,67 @@ func buildSandboxCreateRequestProto(appID, imageID string, params SandboxCreateP
 		idleTimeoutSecs = &v
 	}
 
+	var milliCPU, milliCPUMax *uint32
+	if params.CPU != 0 {
+		if params.CPU <= 0 {
+			return nil, fmt.Errorf("the CPU request (%f) must be a positive number", params.CPU)
+		}
+		v := uint32(1000 * params.CPU)
+		milliCPU = &v
+		if params.CPUMax > 0 {
+			if params.CPUMax < params.CPU {
+				return nil, fmt.Errorf("the CPU request (%f) cannot be higher than CPUMax (%f)", params.CPU, params.CPUMax)
+			}
+			vMax := uint32(1000 * params.CPUMax)
+			milliCPUMax = &vMax
+		}
+	} else if params.CPUMax > 0 {
+		return nil, fmt.Errorf("must also specify CPU request when CPUMax is specified")
```

Comment:
The CPU request can be 0 or empty. I think we can explicitly say it is non-zero:

```suggestion
		return nil, fmt.Errorf("must also specify non-zero CPU request when CPUMax is specified")
```

Same comment for memory.